### PR TITLE
[CDAP-16412] fixing leftover error and refactoring

### PR DESCRIPTION
**Pull request by itsanudeep**
Fixes leftover page level error in hydrator and refactoring namespace resolver.

https://issues.cask.co/browse/CDAP-16412

### CODE REVIEW

File: cdap-ui/app/hydrator/routes.js [hunk 1]
```diff
@@ -128,6 +112,9 @@ angular.module(PKG.name + '.feature.hydrator')
               });
               return defer.promise;
             },
+            rValidNamespace: function($stateParams, myNamespace, myHelpers){
+              myHelpers.validNamespaceResolver($stateParams, myNamespace);
```
Comment (elfenheart) [inline cdap-ui/app/hydrator/routes.js:120]: can you check how many times this function gets called when you go to either studio or detail page? I think all you need to do is add it to the parent `hydrator` state.

File: cdap-ui/app/services/helpers.js [hunk 2]
```diff
@@ -221,6 +221,35 @@ angular.module(PKG.name+'.services')
 
   /* ----------------------------------------------------------------------- */
 
+  function validNamespaceResolver(stateParams, myNamespace) {
+    const { namespace } = stateParams;
+    myNamespace
+      .getList()
```
Comment (elfenheart) [inline cdap-ui/app/services/helpers.js]: this is fetching the namespace right? Can we use the NamespaceStore instead?

File: cdap-ui/app/hydrator/routes.js (hunk 1 above)
Comment (itsanudeep) [reply cdap-ui/app/hydrator/routes.js:120]: For some reason `hydrator` state resolvers are not called when I navigate from detail page to studio page. This is the root cause of the bug.

File: cdap-ui/app/services/helpers.js (hunk 2 above)
Comment (itsanudeep) [reply cdap-ui/app/services/helpers.js]: fixed

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Comment by itsanudeep** [comment]
Build and Test: https://builds.cask.co/browse/CDAP-UDUT537-3
RAT and Checkstyle: https://builds.cask.co/browse/CDAP-DRC9870

**Pull request merged**